src/hexkit/providers/akafka/provider/eventsub.py

```diff
@@ -181,76 +226,220 @@ async def construct(
+            else:
+                raise self.RetriesExhaustedError(
+                    event_type=event.type_, max_retries=self._max_retries
+                ) from err
+
+    async def _handle_consumption(self, *, event: ExtractedEventInfo):
+        """Try to pass the event to the consumer.
+
+        If the event fails:
+        1. Retry until retries are exhausted, if retries are configured.
+        2. Publish the event to the DLQ topic if the DLQ is enabled. Done afterward.
+           or
+        3. Allow failure with unhandled error if DLQ is not configured.
+        """
+        try:
+            await self._translator.consume(
+                payload=event.payload,
+                type_=event.type_,
+                topic=event.topic,
+                key=event.key,
+            )
+        except Exception as underlying_error:
+            logging.warning(
+                "Failed initial attempt to consume event of type '%s' on topic '%s' with key '%s'.",
+                event.type_,
+                event.topic,
+                event.key,
+            )
+
+            if not self._max_retries:
+                if not self._enable_dlq:
+                    raise  # re-raise Exception
+                await self._publish_to_dlq(event=event)
+                return
+
+            # Don't raise RetriesExhaustedError unless retries are actually attempted
+            try:
+                await self._retry_event(event=event, retries_left=self._max_retries)
+            except (self.RetriesExhaustedError, ValueError) as retry_error:
+                # In case of ValueError, proceed with flow because that error is
+                # likely due to a hexkit bug, not a config/event error.
+                # Publish to the DLQ or re-raise the exception
+                logging.warning(retry_error)
+                if not self._enable_dlq:
+                    raise retry_error from underlying_error
+                await self._publish_to_dlq(event=event)
+
+    def _extract_info(self, event: ConsumerEvent) -> ExtractedEventInfo:
+        """Validate the event, returning the extracted info."""
+        event_info = ExtractedEventInfo(event)
+        if event_info.topic == self._retry_topic:
+            event_info.topic = event_info.headers.get(ORIGINAL_TOPIC_FIELD, "")
+            logging.info(
+                "Received previously failed event from topic '%s' for retry.",
+                event_info.topic,
+            )
+        return event_info
+
+    def _validate_extracted_info(self, event: ExtractedEventInfo):
+        """Extract and validate the event, returning the correlation ID and the extracted info."""
+        correlation_id = event.headers.get("correlation_id", "")
+        errors = []
+        if not event.type_:
+            errors.append("type_ is empty")
+        elif event.type_ not in self._types_whitelist:
+            errors.append(f"type_ '{event.type_}' is not in the whitelist")
```

Comment:
```suggestion
            errors.append(f"event type '{event.type_}' is not in the whitelist")
```

Reply:
This and the similar comment above fixed in [e6271fc](https://github.com/ghga-de/hexkit/pull/123/commits/e6271fca629f1a977f3608e7d8554fd3e30e33cb)